ld.lld cannot link with openmpi (with enabled fortran support in openmpi)

I've just discovered via `git bisect` that `950cfca3abb5ba36505ed286f4ab2424bb2cb8f9` (#7795) may cause failure in the configuration phase. This is on my linux machine, which previously used `-fuse-ld=gold`. I have a `spack`-built LLVM in path (loaded via modules) which, I guess, gets detected and causes the flag `-fuse-ld=lld` to appear in the linker line. I see that there's already a check to see that a basic program can compile though... what can we do to make this more robust?

```
-- Performing Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG
-- Performing Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG - Failed
CMake Error at cmake/setup_finalize.cmake:95 (MESSAGE):
  

    Configuration error: Cannot compile a test program with the final set of
    compiler and linker flags:
      CXX flags (DEBUG): -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections
      LD flags  (DEBUG): -Wl,--as-needed -rdynamic -fuse-ld=lld -L/usr/lib/gcc/x86_64-linux-gnu/8/ -lgfortran -ggdb
      LIBRARIES (DEBUG): <path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/boost-1.69.0-gzvs7exfbtdm4abn5ih2tz4x7uhewjv4/lib/libboost_iostreams-mt.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/boost-1.69.0-gzvs7exfbtdm4abn5ih2tz4x7uhewjv4/lib/libboost_serialization-mt.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/boost-1.69.0-gzvs7exfbtdm4abn5ih2tz4x7uhewjv4/lib/libboost_system-mt.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/boost-1.69.0-gzvs7exfbtdm4abn5ih2tz4x7uhewjv4/lib/libboost_thread-mt.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/boost-1.69.0-gzvs7exfbtdm4abn5ih2tz4x7uhewjv4/lib/libboost_regex-mt.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/boost-1.69.0-gzvs7exfbtdm4abn5ih2tz4x7uhewjv4/lib/libboost_chrono-mt.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/boost-1.69.0-gzvs7exfbtdm4abn5ih2tz4x7uhewjv4/lib/libboost_date_time-mt.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/boost-1.69.0-gzvs7exfbtdm4abn5ih2tz4x7uhewjv4/lib/libboost_atomic-mt.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libmuelu-adapters.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libmuelu-interface.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libmuelu.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libifpack2.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libanasazitpetra.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libModeLaplace.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libanasaziepetra.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libanasazi.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libmapvarlib.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libsuplib_cpp.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libsuplib_c.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libsuplib.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libsupes.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libaprepro_lib.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libchaco.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libio_info_lib.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libIonit.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libIotr.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libIohb.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libIogn.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libIovs.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libIoexo_fac.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libIofx.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libIoex.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libIoss.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libnemesis.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libexoIIv2for32.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libexodus_for.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libexodus.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libamesos2.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libbelostpetra.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libbelosepetra.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libbelos.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libml.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libifpack.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libzoltan2.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libamesos.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libgaleri-xpetra.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libgaleri-epetra.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libaztecoo.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libxpetra-sup.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libxpetra.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libepetraext.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libtrilinosss.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libtpetraext.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libtpetrainout.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libtpetra.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libkokkostsqr.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libtpetraclassiclinalg.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libtpetraclassicnodeapi.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libtpetraclassic.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libtriutils.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libzoltan.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libepetra.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libsacado.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libkokkoskernels.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libteuchoskokkoscomm.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libteuchoskokkoscompat.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libteuchosremainder.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libteuchosnumerics.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libteuchoscomm.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libteuchosparameterlist.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libteuchoscore.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libkokkosalgorithms.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libkokkoscontainers.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libkokkoscore.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/trilinos-12.12.1-intry6vbyyffsvsqo76xfzmbf4ymtweg/lib/libgtest.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/matio-1.5.13-5hyqbnpbfjurzwdqfehjuzf37n4i2xac/lib/libmatio.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/netcdf-4.6.2-6k5yvpktnlm7jsa2zib5uytmsah4wotr/lib/libnetcdf.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/mumps-5.1.1-3mmkzqn2ggqh2dw5fdizzdqwhn4fp72c/lib/libdmumps.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/mumps-5.1.1-3mmkzqn2ggqh2dw5fdizzdqwhn4fp72c/lib/libmumps_common.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/mumps-5.1.1-3mmkzqn2ggqh2dw5fdizzdqwhn4fp72c/lib/libpord.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/netlib-scalapack-2.0.2-c4oh6ijycn3ghtybd3xafasg57ahmb3c/lib/libscalapack.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/hypre-2.15.1-5a3ks3pt2zsvssxdvft54vj2iv5zzldp/lib/libHYPRE.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/hdf5-1.10.4-ozy5y574qva5tm4fgseqsm577btcn4t5/lib/libhdf5.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/zlib-1.2.11-dndjvh6fpzdr3thkyvbemikb52dos4ss/lib/libz.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/hdf5-1.10.4-ozy5y574qva5tm4fgseqsm577btcn4t5/lib/libhdf5_hl.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/parmetis-4.0.3-zweeewwr6oxq7fy5b5v5hbsyc3gr5ysr/lib/libparmetis.so;dl;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/suite-sparse-5.3.0-qwizyk56yyfgs64jncyt7wmuca5w7alw/lib/libumfpack.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/suite-sparse-5.3.0-qwizyk56yyfgs64jncyt7wmuca5w7alw/lib/libcholmod.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/suite-sparse-5.3.0-qwizyk56yyfgs64jncyt7wmuca5w7alw/lib/libccolamd.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/suite-sparse-5.3.0-qwizyk56yyfgs64jncyt7wmuca5w7alw/lib/libcolamd.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/suite-sparse-5.3.0-qwizyk56yyfgs64jncyt7wmuca5w7alw/lib/libcamd.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/suite-sparse-5.3.0-qwizyk56yyfgs64jncyt7wmuca5w7alw/lib/libsuitesparseconfig.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/suite-sparse-5.3.0-qwizyk56yyfgs64jncyt7wmuca5w7alw/lib/libamd.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/metis-5.1.0-2og2oed3ucpn7ky77viwsj2pzmzdqjjy/lib/libmetis.so;rt;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/adol-c-develop-2rs6raysyorckdztjkb3tdot4p42xgra/lib64/libadolc.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/p4est-2.0-wry6yho5f766k6crqacoq5kkeyad4v6x/lib/libp4est.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/p4est-2.0-wry6yho5f766k6crqacoq5kkeyad4v6x/lib/libsc.so;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/openblas-0.3.5-vb72x3uulzxakm4iv5sw7oelq5dz7cfc/lib/libopenblas.so;gfortran;m;quadmath;c;<path/to/spack>/opt/spack/linux-ubuntu18.10-x86_64/gcc-8.2.0/intel-tbb-2019.2-rd4a7jt5pbxyremzsayu7qtzv7vwb2jh/lib/libtbb_debug.so

Call Stack (most recent call first):
  cmake/macros/macro_verbose_include.cmake:19 (INCLUDE)
  CMakeLists.txt:132 (VERBOSE_INCLUDE)

-- Configuring incomplete, errors occurred!
```

FWIW here's the relevant part of `CMakeError.log`.
```
ld.lld: error: corrupt input file: version definition index 0 for symbol mpi_fortran_argv_null_ is out of bounds
>>> defined in <path/to/spack>/linux-ubuntu18.10-x86_64/gcc-8.2.0/mumps-5.1.1-3mmkzqn2ggqh2dw5fdizzdqwhn4fp72c/lib/libdmumps.so

ld.lld: error: corrupt input file: version definition index 0 for symbol mpi_fortran_argvs_null_ is out of bounds
>>> defined in <path/to/spack>/linux-ubuntu18.10-x86_64/gcc-8.2.0/mumps-5.1.1-3mmkzqn2ggqh2dw5fdizzdqwhn4fp72c/lib/libdmumps.so

ld.lld: error: corrupt input file: version definition index 0 for symbol mpi_fortran_errcodes_ignore_ is out of bounds
>>> defined in <path/to/spack>/linux-ubuntu18.10-x86_64/gcc-8.2.0/mumps-5.1.1-3mmkzqn2ggqh2dw5fdizzdqwhn4fp72c/lib/libdmumps.so

ld.lld: error: corrupt input file: version definition index 0 for symbol mpi_fortran_unweighted_ is out of bounds
>>> defined in <path/to/spack>/linux-ubuntu18.10-x86_64/gcc-8.2.0/mumps-5.1.1-3mmkzqn2ggqh2dw5fdizzdqwhn4fp72c/lib/libdmumps.so

ld.lld: error: corrupt input file: version definition index 0 for symbol mpi_fortran_weights_empty_ is out of bounds
>>> defined in <path/to/spack>/linux-ubuntu18.10-x86_64/gcc-8.2.0/mumps-5.1.1-3mmkzqn2ggqh2dw5fdizzdqwhn4fp72c/lib/libdmumps.so
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

Source file was:
int main(){ return 0; }
```

--- a/cmake/checks/check_02_compiler_features.cmake
+++ b/cmake/checks/check_02_compiler_features.cmake
@@ -460,7 +460,11 @@ CHECK_CXX_SOURCE_COMPILES(
   DEAL_II_COMPILER_HAS_FUSE_LD_GOLD)
 RESET_CMAKE_REQUIRED()
 
-IF(DEAL_II_COMPILER_HAS_FUSE_LD_LLD)
+#
+# The ld.lld linker is not compatible with GCC. So we only enable it when
+# the Clang compiler is used. See https://github.com/dealii/dealii/issues/7811
+#
+IF(DEAL_II_COMPILER_HAS_FUSE_LD_LLD AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   ADD_FLAGS(DEAL_II_LINKER_FLAGS "-fuse-ld=lld")
 ELSEIF(DEAL_II_COMPILER_HAS_FUSE_LD_GOLD)
   ADD_FLAGS(DEAL_II_LINKER_FLAGS "-fuse-ld=gold")

